binding/Binding/SKTextBlob.cs

```diff
@@ -218,155 +221,199 @@ private void AddPositionedRunInternal (SKPaint font, ushort[] glyphs, SKPoint[]
-					return runbuffer;
+					}
+
+					return new SKRunBuffer (runbuffer, count, textByteCount);
 				}
 			}
 		}
 
-		internal SKTextBlobBuilderRunBuffer AllocateRunHorizontal (SKPaint font, int count, float y, int textByteCount, SKRect? bounds)
+		public SKHorizontalRunBuffer AllocateRunHorizontal (SKPaint font, int count, float y, int textByteCount, SKRect? bounds)
 		{
 			if (font == null)
 				throw new ArgumentNullException (nameof (font));
 
 			using (var lang = new SKString ()) {
 				unsafe {
-					SKTextBlobBuilderRunBuffer runbuffer;
-					if (bounds is SKRect b)
+					SKRunBufferInternal runbuffer;
+					if (bounds is SKRect b) {
 						SkiaApi.sk_textblob_builder_alloc_run_text_pos_h (Handle, font.Handle, count, y, textByteCount, lang.Handle, &b, out runbuffer);
-					else
+					} else {
 						SkiaApi.sk_textblob_builder_alloc_run_text_pos_h (Handle, font.Handle, count, y, textByteCount, lang.Handle, (SKRect*)IntPtr.Zero, out runbuffer);
-					return runbuffer;
+					}
+
+					return new SKHorizontalRunBuffer (runbuffer, count, textByteCount);
 				}
 			}
 		}
 
-		internal SKTextBlobBuilderRunBuffer AllocateRunPositioned (SKPaint font, int count, int textByteCount, SKRect? bounds)
+		public SKPositionedRunBuffer AllocateRunPositioned (SKPaint font, int count, int textByteCount, SKRect? bounds)
 		{
 			if (font == null)
 				throw new ArgumentNullException (nameof (font));
 
 			using (var lang = new SKString ()) {
 				unsafe {
-					SKTextBlobBuilderRunBuffer runbuffer;
-					if (bounds is SKRect b)
+					SKRunBufferInternal runbuffer;
+					if (bounds is SKRect b) {
 						SkiaApi.sk_textblob_builder_alloc_run_text_pos (Handle, font.Handle, count, textByteCount, lang.Handle, &b, out runbuffer);
-					else
+					} else {
 						SkiaApi.sk_textblob_builder_alloc_run_text_pos (Handle, font.Handle, count, textByteCount, lang.Handle, (SKRect*)IntPtr.Zero, out runbuffer);
-					return runbuffer;
+					}
+
+					return new SKPositionedRunBuffer (runbuffer, count, textByteCount);
 				}
 			}
 		}
 	}
 
-	[StructLayout (LayoutKind.Sequential)]
-	internal struct SKTextBlobBuilderRunBuffer
+	public class SKRunBuffer
 	{
-		private IntPtr glyphs;
-		private IntPtr pos;
-		private IntPtr utf8text;
-		private IntPtr clusters;
+		internal readonly SKRunBufferInternal internalBuffer;
+
+		internal SKRunBuffer (SKRunBufferInternal buffer, int size, int textSize)
+		{
+			internalBuffer = buffer;
+			Size = size;
+			TextSize = textSize;
+		}
+
+		public int Size { get; }
 
-		public IntPtr GlyphsBuffer => glyphs;
+		public int TextSize { get; }
 
-		public IntPtr PositionsBuffer => pos;
+		public void SetGlyphs (IReadOnlyList<ushort> glyphs)
+		{
+			if (glyphs.Count > Size)
+				throw new ArgumentOutOfRangeException (nameof (glyphs));
 
-		public IntPtr Utf8textBuffer => utf8text;
+			var glyphsSpan = GetGlyphSpan ();
 
-		public IntPtr ClustersBuffer => clusters;
+			for (var i = 0; i < Size; i++) {
+				glyphsSpan[i] = glyphs[i];
```

Comment:
What is the performance of this manual copy? Could we use unsafe to copy?

Reply:
We could modify the sk_textblob_builder_runbuffer_set_xxx to take a pointer and the offset and length to copy. Then we could do a memory copy if it is faster. In fact, how often will this set be a partial set? Would it be useful to expose a public api for copying only a subset?

Reply:
I don't think you ever want to write a partial set. You allocate a buffer and want to write to it the full length. This can be done on native side for `public void SetGlyphs (ReadOnlySpan<ushort> glyphs)`

Reply:
It makes sense to have multiple overloads

Reply:
This is how I use it:

```
using (var builder = new SKTextBlobBuilder())
{
    var runBuffer = builder.AllocateRunPositioned(_paint,
    textRun.GlyphRun.GlyphIndices.Length, 0, null);
    var glyphs = runBuffer.GetGlyphSpan();
    var positions = runBuffer.GetPositionSpan();

    for (var i = 0; i < textRun.GlyphRun.GlyphIndices.Length; i++)
    {
        glyphs[i] = textRun.GlyphRun.GlyphIndices[i];
        positions[i] = textRun.GlyphRun.GlyphPositions[i];
    }

    var blob = builder.Build();
    canvas.DrawText(blob, 0, 0, _paint);
}
```

Reply:
In that case, we can use the native method, since that is what is does. But, it might be the same thing to copy in unsafe. Need to run some profiling quickly.

Reply:
Updated this to accept a Span<T> and then use the CopyTo member to just move the memory rather than looping through each item.